ruby.rb

```diff
@@ -3,49 +3,51 @@ class ArticlesController < ApplicationController
 
   def index
     @articles = Article.all.includes(:user)
+    articles_type
+    boundary = articles_boundary(0, 20)
+    offset = boundary[offset]
+    limit = boundary[limit]
+    @articles = @articles.order(created_at: :desc).offset(offset).limit(limit)
+  end
 
+  def articles_type
     if params[:tag].present?
       @articles = @articles.tagged_with(params[:tag])
     elsif params[:author].present?
       @articles = @articles.authored_by(params[:author])
     elsif params[:favorited].present?
       @articles = @articles.favorited_by(params[:favorited])
     end
+  end
 
+  def articles_boundary(offset = nil, limit = nil)
     @articles_count = @articles.count
-    offset = 0
-    limit = 20
-    if params[:offset]
-      offset = params[:offset]
-    end
-    if params[:limit]
-      limit = params[:limit]
-    end
-    @articles = @articles.order(created_at: :desc).offset(offset).limit(limit)
+    offset = params[:offset] if params[:offset]
+    limit = params[:limit] if params[:limit]
+
+    { offset: offset, limit: limit }
   end
 
   def feed
-    @articles = Article.includes(:user).where(user: current_user.following_users)
+    @articles = Article
+                .includes(:user)
+                .where(user: current_user.following_users)
 
-    @articles_count = @articles.count
-    if params[:offset]
-      offset = params[:offset]
-    end
-    if params[:limit]
-      limit = params[:limit]
-    end
+    boundary = articles_boundary
+    offset = boundary[offset]
+    limit = boundary[limit]
     @articles = @articles.order(created_at: :desc).offset(offset).limit(limit)
     render :index
   end
 
   def create
-    @article = Article.new(articleParams)
+    @article = Article.new(article_params)
     @article.user = current_user
```

Comment:
The definition of article params can merge user so you dont have to do this. e.g:

```
def article_params
  params.require(:article).permit(:title, :body, :description, tag_list: []).
     merge(user: current_user)
end
```